reflect.go

```diff
@@ -831,7 +873,91 @@ func (r *Reflector) typeName(t reflect.Type) string {
 		}
 	}
 	if r.FullyQualifyTypeNames {
-		return t.PkgPath() + "." + t.Name()
+		return fullyQualifiedTypeName(t)
 	}
 	return t.Name()
 }
+
+func fullyQualifiedTypeName(t reflect.Type) string {
+	return t.PkgPath() + "." + t.Name()
+}
+
+// AddGoComments will read all the go files contained in the provided path,
+// including sub-directories, in order to generate a dictionary of comments
+// associated with Types and Fields. The results will be added to the `CommentsMap`
+// to be included in Schema "description" fields.
+//
+// The `go/parser` library is used to extract all the comments and unfortunately doesn't
+// have a built-in way to determine the fully qualified name of a package. The `base` paremeter,
+// the URL used to import that package, is thus required to be able to match reflected types.
+//
+// When parsing type comments, we use the `go/doc`'s Synopsis method to extract the first phrase
+// only. Field comments, which tend to be much short, will include everything.
+func (r *Reflector) AddGoComments(base, path string) error {
```

Comment:
I think it would be good to move this to a new file, this is getting quite long.

Reply:
Agreed. I'll see what I can do.